test/router.spec.js

```diff
@@ -592,6 +592,216 @@ describe('Restify Router', function () {
     });
   });
 
+  describe('Nested routers via .group', function () {
+    it('Should allow arbitrary nesting of routers', function (done) {
+      var router = new Router();
+
+      router.group('/v1', function (router) {
+        router.group('/auth', function (router) {
+          router.post('/register', function (req, res, next) {
+            res.send({
+              status: 'success',
+              name: req.body.name
+            });
+            return next();
+          });
+        });
+      });
+
+      router.applyRoutes(server);
+
+      request(server)
+        .post('/v1/auth/register')
+        .set('Content-Type', 'application/json')
+        .send({name: 'test'})
+        .expect(200)
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+
+          res.body.should.deep.equal({
+            status: 'success',
+            name: 'test'
+          });
+          done();
+        });
+
+    });
+
+    it('Should allow nesting routers using regex paths', function (done) {
+      var router = new Router();
+
+      router.group(/^\/foo+/, function (router) {
+        router.group(/^\/ba+r/, function (router) {
+          router.get('/bam', function (req, res, next) {
+            res.send(200);
+            return next();
+          });
+        });
+      });
+
+      router.applyRoutes(server);
+
+      request(server)
+        .get('/fooo/baar/bam')
+        .expect(200)
+        .end(function (err) {
+          if (err) {
+            return done(err);
+          }
+          done();
+        });
+
+    });
+
+    it('Should fail if invalid path type is provided for router', function () {
+
+      function fail() {
+        var router = new Router();
+        var nested = new Router();
```

Comment:
is this used?

Reply:
This variable has been forgotten, it is not used =P

I updated the fork with the master and removed this variable.